grafana-plugin/src/containers/AddResponders/parts/AddRespondersPopup/AddRespondersPopup.tsx

```diff
@@ -97,11 +97,17 @@ const AddRespondersPopup = observer(
     );
 
     const searchForUsers = useCallback(async () => {
-      /**
-       * specifying is_currently_oncall=all will tell the backend not to paginate the results
-       */
-      const userResults = await userStore.search<UserCurrentlyOnCall[]>({ searchTerm, is_currently_oncall: 'all' });
-      setUserSearchResults(userResults);
+      const _searchForUsers = (is_currently_oncall: string) =>
+        userStore.search<PaginatedUsersResponse<UserCurrentlyOnCall>>({ searchTerm, is_currently_oncall });
+
+      const onCallUserSearchResultsPromise = _searchForUsers('true').then((resp) =>
+        setOnCallUserSearchResults(resp.results)
+      );
+      const notOnCallUserSearchResultsPromise = _searchForUsers('false').then((resp) =>
+        setNotOnCallUserSearchResults(resp.results)
+      );
+
+      await Promise.all([onCallUserSearchResultsPromise, notOnCallUserSearchResultsPromise]);
     }, [searchTerm]);
```

Comment:
Right now we have many places where we mix usage of `async/await` and `then()` but this is something that it would be good to unify in nearest future and switch fully to `async/await`. Before we do it across the whole codebase, maybe we can start to use `async/await` in new code?  

Here we can probably achieve exactly same result with:
```
const searchForUsers = useCallback(async () => {
      const _search = async (is_currently_oncall: 'true' | 'false' | 'all') => {
        const response = await userStore.search<PaginatedUsersResponse<UserCurrentlyOnCall>>({
          searchTerm,
          is_currently_oncall,
        });
        return response.results;
      };

      const [onCallUserSearchResults, notOnCallUserSearchResults] = await Promise.all([
        _search('true'),
        _search('false'),
      ]);

      setOnCallUserSearchResults(onCallUserSearchResults);
      setNotOnCallUserSearchResults(notOnCallUserSearchResults);
    }, [searchTerm]);
    ```
    
    What do you think?

Reply:
great point, thanks for the suggestion. updated this 👍